src/chart.ts

```diff
@@ -635,18 +635,75 @@ export class Chart extends EventEmitter.EventEmitter {
   updateAxes() {
     const instance = this
     const canvas = instance.canvas.merge(instance.canvas.enter)
-    canvas.select('.x.axis').call(instance.meta.xAxis)
-    canvas.select('.y.axis').call(instance.meta.yAxis)
 
-    // updates the style of the axes
+    // center the axes
+    const yMin = this.meta.yScale.domain()[0]
+    const yMax = this.meta.yScale.domain()[1]
+
+    const yMid = (yMax + yMin) / 2
+
+    const yScaleFactor = this.meta.height / (yMax - yMin)
+
+    let yTranslation = yScaleFactor * yMid + this.meta.height / 2
+
+    const xMin = this.meta.xScale.domain()[0]
+    const xMax = this.meta.xScale.domain()[1]
+
+    const xMid = (xMax + xMin) / 2
+
+    const xScaleFactor = this.meta.width / (xMin - xMax)
+
+    let xTranslation = xScaleFactor * xMid + this.meta.width / 2
+
+    if (this.options.xAxis.position === 'sticky') {
+      yTranslation = yTranslation < 0 ? 0 : yTranslation
+      yTranslation = yTranslation > this.meta.height ? this.meta.height : yTranslation
+      canvas
+        .select('.x.axis')
+        .call(instance.meta.xAxis)
+        .attr('transform', 'translate(0,' + yTranslation + ')')
```

Comment:
I see this line in two places:

```
        .select('.x.axis')
        .attr('transform', 'translate(0,' + yTranslation + ')')

...

        .selectAll('.x.axis path, .x.axis line')
        .attr('transform', 'translate(0,' + yTranslation + ')')
```

Wouldn't the first one automatically apply a translation to the entire group (including the `path` and `line`)?

Reply:
No, those are different values. I don't know why, but when I draw the x lines only with yTranslation, it's buggy, and the grid doesn't render correctly.

`
        .attr('transform', 'translate(0,' + (this.meta.height / 2 - yTranslation + this.meta.height / 2) + ')');
`

I use this for all Paths and Lines